### PR TITLE
Fix leaking goroutines in scheduler integration tests

### PR DESCRIPTION
**Pull request by wojtek-t**
Ref https://github.com/kubernetes/kubernetes/issues/108483

```release-note
NONE
```

/kind cleanup
/priority important-longterm
/sig scheduling

### CODE REVIEW

**Comment by wojtek-t** [comment]
@aojea - this is now ready for review

File: test/integration/framework/test_server.go [hunk 1]
```diff
@@ -59,7 +60,7 @@ type TearDownFunc func()
 
 // StartTestServer runs a kube-apiserver, optionally calling out to the setup.ModifyServerRunOptions and setup.ModifyServerConfig functions
 func StartTestServer(t *testing.T, setup TestServerSetup) (client.Interface, *rest.Config, TearDownFunc) {
-	certDir, err := os.MkdirTemp("", "test-integration-"+t.Name())
+	certDir, err := os.MkdirTemp("", "test-integration-"+strings.ReplaceAll(t.Name(), "/", "_"))
```
Comment (aojea) [inline test/integration/framework/test_server.go:63]: Ugh 😩

Comment (wojtek-t) [reply test/integration/framework/test_server.go:63]: It took me a bit to debug it - for tests that have subcases using t.Run(...), their names contain "/", and for those creating directory was failing - seems like the simplest fix.

**Pull request merged**